### DISCUSSION TITLE
Selective deployments/pull request

### DISCUSSION THREAD

**Discussion by EnglundAndreas** [hardisgroupcom/sfdx-hardis#548] [Q&A] [answered]
When we do a PR from preprod to merge into master to deploy to production, all files/commits are included. Not everything in preprod may be approved for prod yet, but some features/fixes that have been tested and approved should still be deployed. 

Is there any easy way to only include some commits/files from preprod to prod/master, like a selective PR?

**Comment by nvuillam** [comment, marked as answer]
@EnglundAndreas when your Application Lifecycle Management is with CI/CD, you don't deploy features, you deploy branches that contain features :)

So selective deployment is not an option, because multiple features can have dependencies between each other, and when you test, you test the whole state of a branch, not standalone metadatas, so if you don''t deploy the full version, what you tested before might not work in target org 

A classic way to handle urgent - not urgent deployments is to handle 2 streams: RUN & BUILD
- features that needs to be tested: PR with integration branch as target
- features for RUN (hotfixes): PR with preprod branch as target
- RUN merged PRs must be periodically retrofitted in integration to avoid overwriting of RUN metadatas by BUILD metadatas
![image](https://github.com/hardisgroupcom/sfdx-hardis/assets/17500430/c0323e7a-a1a4-4135-a85c-18782e395717)

If you really want to deploy in production something that is in uat, you can eventually "move" features from BUILD to RUN stream, by "cherry picking" the commits from uat in a new branch under preprod, then PR to preprod, then PR preprod -> main
